internal/cloudapi/v2/imagerequest.go

```diff
@@ -33,57 +33,232 @@ func (ir *ImageRequest) GetImageOptions(imageType distro.ImageType, bp blueprint
+	return t, nil
+}
+
+func newAzureTarget(ir *ImageRequest, imageType distro.ImageType) (*target.Target, error) {
+	var azureUploadOptions AzureUploadOptions
+	jsonUploadOptions, err := json.Marshal(*ir.UploadOptions)
+	if err != nil {
+		return nil, HTTPError(ErrorJSONMarshallingError)
+	}
+	err = json.Unmarshal(jsonUploadOptions, &azureUploadOptions)
+	if err != nil {
+		return nil, HTTPError(ErrorJSONUnMarshallingError)
+	}
+	rgLocation := ""
+	if azureUploadOptions.Location != nil {
+		rgLocation = *azureUploadOptions.Location
+	}
+	t := target.NewAzureImageTarget(&target.AzureImageTargetOptions{
+		TenantID:       azureUploadOptions.TenantId,
+		Location:       rgLocation,
+		SubscriptionID: azureUploadOptions.SubscriptionId,
+		ResourceGroup:  azureUploadOptions.ResourceGroup,
+	})
+
+	if azureUploadOptions.ImageName != nil {
+		t.ImageName = *azureUploadOptions.ImageName
+	} else {
+		// if ImageName wasn't given, generate a random one
+		t.ImageName = fmt.Sprintf("composer-api-%s", uuid.New().String())
+	}
+	t.OsbuildArtifact.ExportFilename = imageType.Filename()
+	return t, nil
+}
+
+func newOCITarget(ir *ImageRequest, imageType distro.ImageType) (*target.Target, error) {
+	var ociUploadOptions OCIUploadOptions
+	jsonUploadOptions, err := json.Marshal(*ir.UploadOptions)
+	if err != nil {
+		return nil, HTTPError(ErrorJSONMarshallingError)
+	}
+	err = json.Unmarshal(jsonUploadOptions, &ociUploadOptions)
+	if err != nil {
+		return nil, HTTPError(ErrorJSONUnMarshallingError)
+	}
+
+	key := fmt.Sprintf("composer-api-%s", uuid.New().String())
+	t := target.NewOCIObjectStorageTarget(&target.OCIObjectStorageTargetOptions{})
+	t.ImageName = key
+	t.OsbuildArtifact.ExportFilename = imageType.Filename()
+	return t, nil
+}
+
+func newPulpOSTreeTarget(ir *ImageRequest, imageType distro.ImageType) (*target.Target, error) {
+	var pulpUploadOptions PulpOSTreeUploadOptions
+	jsonUploadOptions, err := json.Marshal(*ir.UploadOptions)
+	if err != nil {
+		return nil, HTTPError(ErrorJSONMarshallingError)
+	}
+	err = json.Unmarshal(jsonUploadOptions, &pulpUploadOptions)
+	if err != nil {
+		return nil, HTTPError(ErrorJSONUnMarshallingError)
+	}
+	serverAddress := ""
+	if pulpUploadOptions.ServerAddress != nil {
+		serverAddress = *pulpUploadOptions.ServerAddress
+	}
+	repository := ""
+	if pulpUploadOptions.Repository != nil {
+		repository = *pulpUploadOptions.Repository
+	}
+	t := target.NewPulpOSTreeTarget(&target.PulpOSTreeTargetOptions{
+		ServerAddress: serverAddress,
+		Repository:    repository,
+		BasePath:      pulpUploadOptions.Basepath,
+	})
+
+	t.ImageName = fmt.Sprintf("composer-api-%s", uuid.New().String())
+	t.OsbuildArtifact.ExportFilename = imageType.Filename()
+	return t, nil
+}
+
+// Returns the name of the default target for a given image type name or error
+// if the image type name is unknown.
+func getDefaultTarget(imageType ImageTypes) (ImageRequestUploadTarget, error) {
+	switch imageType {
```

Comment:
Almost seems like this doesn't need to be a switch anymore, a map would also work.

Reply:
yeah, I considered it.  I did the map for the support mapping because that felt easier, but this was already there so I didn't want to rewrite it.  But maybe it's worth the consistency.